bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/PJLinkDevice.java

```diff
@@ -230,11 +231,11 @@ public synchronized String execute(String command) throws IOException, Authentic
             logger.debug("Got empty string response for request '{}' from {}, waiting for another line", response,
                     fullCommand.replaceAll("\r", "\\\\r"));
         }
-        logger.debug("Got response '{}' for request '{}' from {}", response, fullCommand.replaceAll("\r", "\\\\r"),
-                ipAddress.toString());
         if (response == null) {
             throw new ResponseException("Response to request '" + fullCommand.replaceAll("\r", "\\\\r") + "' was null");
         }
+        logger.debug("Got response '{}' ({}) for request '{}' from {}", response, Arrays.toString(response.getBytes()),
```

Comment:
It's always good to have more details!

The string operations are relatively expensive so it's better to make the logging conditional using `logger.isDebugEnabled()` . Another improvement is to always remove `.toString()` from all log message parameters because this it will automatically be called by the logging framework but only when the logging is enabled.

Reply:
Thanks @wborn, I just did the adjustment :+1: